FYI: .Net 4 support, VS2010, all kind of dependency upgrades, NuGet

cdrnet opened the pull request:
I've quickly upgraded the whole project to VS 2010 and .NET 4, and almost all its dependencies to the newest versions. Switched to NuGet references for Spark, StructureMap and RhinoMocks.

Note that this upgrade is barely tested, I only ensured that it works fine for the .Net 4 assembly which I wanted to generate documentation for: MathNet.Numerics. Therefore, please consider this pull request only as a "For your Information" note, pulling at your own risk.

Thanks,
Chris

cdrnet committed:
UPGRADE to .Net 4.0 and VS2010, now supporting building docs for .Net 4 assemblies. Upgraded to Spark 1.5 (NuGet), StructureMap 2.6.1 (NuGet), RhinoMocks 3.6 (NuGet), NUnit 2.5.9, ILMerge 2.10

jagregory commented:
Awesome, much appreciated. This needed doing, so even untested this will be useful.

Martin-Bohring committed:
Upgraded NuGet package dependencies to the latest versions

Martin-Bohring committed:
Removed Nuget package update leftovers

Martin-Bohring committed:
Updated build project to account for the updated packages

(I need to find a better way to deal with package updates)

Martin-Bohring committed:
Changed to NUnit references to NuGet package references

Updated to the newest version along the way
Changed the build to use the NUnit test runner from the NUnit.Runners NuGet package
Updated Machine specifications to a NuGet reference and to the 0.5.10 version

Martin-Bohring committed:
Forgot to check in the updated project files

(VS did not save them after change for whatever reasons)

Martin-Bohring committed:
Removed no more needed Machine.Specification in the lib folder, because the NuGet package reference is used by now

Martin-Bohring committed:
Added MSBuild community task as NuGet reference

Martin-Bohring committed:
Updated build to use the MSBuild community task NuGet package reference

Martin-Bohring committed:
Removed MSBuild community tasks from lib folder

Martin-Bohring committed:
Added ILMerge NuGet package

Martin-Bohring committed:
Fixed not using the already defined variable within the build to import MSBuild community targets

Martin-Bohring committed:
Fixed relative path error in build project

Martin-Bohring committed:
Removed no more needed outdated ILMerge tool, because the NuGet package is used by now

Commit:
Removed not needed assembly references

Commit:
Started minor code cleanups

Commit:
Started some StyleCop and FxCop cleanup

Commit:
Another round of StyleCop fixes to get acquainted with the code

Commit:
Another round of code cleanup

(The intention is to introduce some dogfooding by adding XML doc to docu itself)

Commit:
More cleanup

Martin-Bohring committed:
Tidied up the formatting

Martin-Bohring committed:
Removed no needed and outdated bootstrapper package definitions

Martin-Bohring committed:
Changed MSBuild Community Task import back to the "Build" sub directory

(Apparently the directory id created when adding the MSBuild Community Task NuGet package.
That could have been documented in their readme and made more transparent, but otherwise a clever hack)

Commit:
CRLF problems

Fixed StyleCop warnings

Commit:
Added XML documentation to enable more DogFooding

(Fixed StyleCop warnings)

Commit:
CRLF fixes

lukaskabrt committed:
Implemented Constructor support

cdrnet committed:
Add proper .gitattributes file

cdrnet committed:
Upgrade all NuGet packages

cdrnet committed:
Do not ignore ref and out method arguments

cdrnet committed:
Rename 'Undocumented' to 'Reflected' for clarity

cdrnet committed:
Fix bug that caused tons of methods not to be documented (because a method with the same name exists in another type)

cdrnet committed:
Fixing line endings (should no longer be a problem since we now have a proper .gitattributes)

cdrnet committed:
Upgrade to .Net 4.5, warning: changes ordering

cdrnet committed:
Simplify

cdrnet committed:
Simplify: drop IoC (no longer hide complexity)

cdrnet committed:
Simplify: ConsoleApplication.Run

cdrnet committed:
Simplify: drop IScreenWriter/IScreenMessage abstraction (WTF)

cdrnet committed:
Simplify: no need for switches abstraction

cdrnet committed:
Simplify: no need for ConsoleApplication abstraction

cdrnet committed:
Simplify: generator, output, drop redundant interfaces

cdrnet committed:
Simplify: no need for full-blown domain events

cdrnet committed:
Simplify: comment object model no need for interface

cdrnet committed:
Cosmetics: generators

cdrnet committed:
Cosmetics: documentation

cdrnet committed:
Refactoring: model builder

cdrnet committed:
Simplify: model generation steps

cdrnet committed:
Cosmetics: generators

cdrnet committed:
Simplify: untangle Identifier class

cdrnet committed:
Fetch xml documentation also for inherited members (was missing)

cdrnet committed:
Track obsolete members

cdrnet committed:
Warn on obsolete types and members, separate summary of static methods, do not render non-static methods on static types

cdrnet committed:
Support for generic method parameters #1

cdrnet committed:
Support for generic array method parameters #1

cdrnet committed:
Show static functions before instance methods
